### PR TITLE
Pass initial population to EpiSystem, not EpiEnv

### PR DESCRIPTION
**Pull request by sdl1**
This is a pre-cursor to #54 

This removes the `initial_population` member from the `EpiEnv` struct. Instead, the `initial_population` can be passed to an `EpiSystem` constructor, where it is used to initialise the abundances.

There are a few reasons for this:
- It is conceptually cleaner and means we can easily support constructing an `EpiSystem` without an `initial_population`
- It means that in #54, we don't have to make any changes to `EpiEnv`, instead, we just have some small changes in `EpiLandscape` and `EpiSystem`
- Also in #54, it will be possible to make `EpiLandscape.grid` the appropriate `AxisArray` in the `EpiSystem` constructor if `initial_population` is not passed in. This is hard to do in the current code, because `initial_population` is set to a default matrix of zeros in `EpiEnv`, which does not match with the actual shape of the classes

So basically with this PR the workflow is
```julia
total_pop = # load population
total_pop.data[total_pop .≈ 0.0] .= NaN
# Explicitly call the shrink function, which shrinks based on the NaNs
# We could easily wrap all this in a separate function
total_pop = shrink_to_active(total_pop)
...

# Just pass in the size here, like it used to be
epienv = simplehabitatAE(298.0K, size(total_pop), area, NoControl())
...

# Pass in the total_pop here. This will also take care of modifying the active region based on the NaNs in total_pop
epi = EpiSystem(epilist, epienv, rel, total_pop, UInt16(1))
```

Some of the diffs are large but this is mostly just moving around code

### CODE REVIEW

**Review by claireh93 — APPROVED** [review]
This looks great @sdl1! Ready to merge
